include/nbl/builtin/hlsl/emulated/float64_t.hlsl

```diff
@@ -0,0 +1,657 @@
+                uint64_t mantissa = emulated_float64_t_impl::divmod128by64(lhsMantissaShifted.x, lhsMantissaShifted.y, rhsRealMantissa);
+
+                const int msb = emulated_float64_t_impl::_findMSB(mantissa);
+                if(msb != -1)
+                {
+                    const int shiftAmount = 52 - msb;
+                    assert(shiftAmount >= 0);
+                    mantissa <<= shiftAmount;
+                    exp -= shiftAmount;
+                }
+
+                mantissa &= ieee754::traits<float64_t>::mantissaMask;
+
+                return bit_cast<this_t>(emulated_float64_t_impl::assembleFloat64(sign, uint64_t(exp) << ieee754::traits<float64_t>::mantissaBitCnt, mantissa));
+            }
+            else
+            {
+                //static_assert(false, "not implemented yet");
+                return bit_cast<this_t>(0xdeadbeefbadcaffeull);
+            }
+        }
+
+        emulated_float64_t operator/(const float rhs) NBL_CONST_MEMBER_FUNC
+        {
+            return bit_cast<this_t>(data) / create(rhs);
+        }
+
+        // relational operators
+        bool operator==(this_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath)
+            {
+                if (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data))
+                    return false;
+                if (emulated_float64_t_impl::areBothZero(data, rhs.data))
+                    return true;
+            }
+
+            const emulated_float64_t xored = bit_cast<this_t>(data ^ rhs.data);
+            if ((xored.data & 0x7FFFFFFFFFFFFFFFull) == 0ull)
+                return true;
+
+            return !(xored.data);
+        }
+        bool operator!=(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath && (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data)))
+                return false;
+
+            return !(bit_cast<this_t>(data) == rhs);
+        }
+        bool operator<(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath)
+            {
+                if (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data))
+                    return false;
+                if (emulated_float64_t_impl::areBothInfinity(data, rhs.data))
+                    return false;
+                if (emulated_float64_t_impl::areBothZero(data, rhs.data))
+                    return false;
+            }
+
+            const uint64_t lhsSign = ieee754::extractSign(data);
+            const uint64_t rhsSign = ieee754::extractSign(rhs.data);
+
+            // flip bits of negative numbers and flip signs of all numbers
+            uint64_t lhsFlipped = data ^ ((0x7FFFFFFFFFFFFFFFull * lhsSign) | ieee754::traits<float64_t>::signMask);
+            uint64_t rhsFlipped = rhs.data ^ ((0x7FFFFFFFFFFFFFFFull * rhsSign) | ieee754::traits<float64_t>::signMask);
```

Comment:
something tells me you can perform this re-arrangement more quickly without resorting to integer mul (its not fast)

maybe try `^(sign ?  0x7fFFffFFffFFffFFull:0x0ull)` or just `if (sign) data ^= 0x7fFFffFFffFFffFFull;` and do the comparison as `int64_t` 

OR `^(sign ?  0x7fFFffFFffFFffFFull:0x8000000000000000ull)` and `uint64_t` comp